### PR TITLE
Backport updated ecalBadCalibFilter crystal list [102X]

### PR DESCRIPTION
**Pull request by peruzzim**
backport #25436 

The update of the crystal list is protected by era modifiers to avoid changing MiniAOD content.
Depends on #25444.

@lathomas

### CODE REVIEW

**Comment by peruzzim** [comment]
please test with #25444

**Comment by peruzzim** [comment]
please abort test

**Comment by peruzzim** [comment]
please test with #25444

File: RecoMET/METFilters/python/ecalBadCalibFilter_cfi.py [hunk 1]
```diff
@@ -30,3 +30,20 @@
...
+from Configuration.Eras.Modifier_run2_nanoAOD_94XMiniAODv1_cff import run2_nanoAOD_94XMiniAODv1
+from Configuration.Eras.Modifier_run2_nanoAOD_94XMiniAODv2_cff import run2_nanoAOD_94XMiniAODv2
+from Configuration.Eras.Modifier_run2_nanoAOD_102Xv1_cff import run2_nanoAOD_102Xv1
+for modifier in run2_miniAOD_devel, run2_nanoAOD_94XMiniAODv1, run2_nanoAOD_94XMiniAODv2, run2_nanoAOD_102Xv1:
+    modifier.toModify(ecalBadCalibFilter, baddetEcal = [872439604,872422825,872420274,872423218,
+                                                        872423215,872416066,872435036,872439336,
+                                                        872420273,872436907,872420147,872439731,
+                                                        872436657,872420397,872439732,872439339,
+                                                        872439603,872422436,872439861,872437051,
+                                                        872437052,872420649,872421950,872437185,
```
Comment (slava77) [inline RecoMET/METFilters/python/ecalBadCalibFilter_cfi.py]: please do not copy-paste already existing IDs from ```run2_common```.
This part here should contain just the additions.
Consider also that the common nanoAOD overrides may better be grouped into one modifier.


Also, a recommended syntax for multiple modifiers is using ```(a | b).toModify```
So, ```(run2_miniAOD_devel | run2_nanoAOD_94XMiniAODv1 |  run2_nanoAOD_94XMiniAODv2 | run2_nanoAOD_102Xv1).toModify

File: RecoMET/METFilters/python/ecalBadCalibFilter_cfi.py [hunk 2]
```diff
@@ -30,3 +30,20 @@
...
+                                                        872422564,872421566,872421695,872421955,
+                                                        872421567,872437184,872421951,872421694,
+                                                        872437056,872437057,872437313])
+for modifier in run2_nanoAOD_94XMiniAODv1, run2_nanoAOD_94XMiniAODv2, run2_nanoAOD_102Xv1:
+    modifier.toModify(ecalBadCalibFilter, EcalRecHitSource = cms.InputTag('reducedEgamma','reducedEERecHits'))
```
Comment (slava77) [inline RecoMET/METFilters/python/ecalBadCalibFilter_cfi.py]: please drop type specifications for all parameters which already exist.
This is a safer way to protect from parameter name mistakes and will also help in possible parameter migrations.

File: RecoMET/METFilters/python/ecalBadCalibFilter_cfi.py (hunk 1 above)
Comment (peruzzim) [reply RecoMET/METFilters/python/ecalBadCalibFilter_cfi.py]: How can I extend the list?

File: RecoMET/METFilters/python/ecalBadCalibFilter_cfi.py (hunk 2 above)
Comment (peruzzim) [reply RecoMET/METFilters/python/ecalBadCalibFilter_cfi.py]: Do you mean `EcalRecHitSource = ('reducedEgamma','reducedEERecHits')`?

File: RecoMET/METFilters/python/ecalBadCalibFilter_cfi.py (hunk 1 above)
Comment (slava77) [reply RecoMET/METFilters/python/ecalBadCalibFilter_cfi.py]: ```+= [a, b, c]``` ?

File: RecoMET/METFilters/python/ecalBadCalibFilter_cfi.py (hunk 2 above)
Comment (slava77) [reply RecoMET/METFilters/python/ecalBadCalibFilter_cfi.py]: right.
IIUC, ```'reducedEgamma:reducedEERecHits'``` should work as well

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by peruzzim** [comment]
please abort test

needs testing with #25444 + I took the chance to also fix the modifier syntax as suggested

**Comment by peruzzim** [comment]
please test with #25444

**Comment by peruzzim** [comment]
please test with #25444

**Comment by fabiocos** [comment]
+1

physics changes are restricted to NanoAOD and miniAOD development modifier, the default stays unchanged

**Comment by fabiocos** [comment]
merge

**Pull request merged**